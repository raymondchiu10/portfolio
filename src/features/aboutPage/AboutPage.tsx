import profile from "src/images/jpg/arizona_profile.jpg";
import profilewebp from "src/images/webp/arizona_profile.webp";

const AboutPage = () => {
	const startDate = 2018;
	const date = new Date().getFullYear();

	return (
		<section className="aboutPage">
			<div className="aboutPage_profile">
				<div className="aboutPage_profile_container">
					<picture>
						<source srcSet={profilewebp} type="image/webp" />
						<source srcSet={profile} type="image/jpeg" />
						<img src={profile} alt="Raymond Chiu" />
					</picture>
				</div>
			</div>

			<h2>Hey there</h2>
			<div className="aboutPage_body">
				<p>
					{
						"I'm a Full Stack Developer with a background in Digital Design and Development, user experience/interface design with a focus on Frontend Development."
					}
				</p>

				<p>{`I have been building my craft in the industry over the last ${Number(date - startDate)}+ years and has had the privilege of working with exceptional companies, such as Frame One Software, Expopulus, and Xai Games, and many others. I also have experience working with startups and projects such as Netcoins, ShackHacker, and Revyva.`}</p>

				<p>
					{
						"I also enjoy playing disc golf, video games, and enjoying the local restraunts in the area."
					}
				</p>

				<p>
					{
						"You may have noticed that almost all my work is in screenshot form - almost all of my professional work is on private repositories, so if you would like to know more, shoot me a message on my "
					}
					<a
						href="https://www.linkedin.com/in/rachiu/"
						target="_blank"
						rel="noopener noreferrer"
					>
						{"LinkedIn"}
					</a>
				</p>
			</div>
		</section>
	);
};

export default AboutPage;
