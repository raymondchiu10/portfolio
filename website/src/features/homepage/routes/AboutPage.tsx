export function AboutPage() {
	return (
		<section className="aboutPage">
		</section>
	);
}