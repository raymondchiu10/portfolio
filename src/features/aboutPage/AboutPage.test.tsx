import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import AboutPage from "./AboutPage";

describe("About Page", () => {
	it("renders the page", () => {
		const { asFragment } = render(<AboutPage />);
		expect(asFragment()).toMatchSnapshot();
	});

	it("renders the profile image", () => {
		render(<AboutPage />);
		const imageElement = screen.getByAltText("Raymond Chiu");
		expect(imageElement).toBeInTheDocument();
	});

	it("renders the CTA link", () => {
		render(<AboutPage />);
		const linkElement = screen.getByText("LinkedIn");
		expect(linkElement).toBeInTheDocument();
	});
});
