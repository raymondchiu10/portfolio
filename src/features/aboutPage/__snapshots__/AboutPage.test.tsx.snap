// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`about page render > renders the page 1`] = `
<DocumentFragment>
  <section
    class="aboutPage"
  >
    <div
      class="aboutPage_profile"
    >
      <div
        class="aboutPage_profile_container"
      >
        <img
          alt="Raymond Chiu"
          src="/src/images/arizona_profile.jpg"
        />
      </div>
    </div>
    <h2>
      Hey there,
    </h2>
    <div
      class="aboutPage_body"
    >
      <p>
        I'm a Full Stack Developer with a background in Digital Design and Development, user experience/interface design with a focus on Frontend Development.
      </p>
      <p>
        I have been building my craft in the industry over the last 6+ years and has had the privilege of working with exceptional companies, such as Frame One Software, Expopulus, and Xai Games, and many others. I also have experience working with startups and projects such as Netcoins, ShackHacker, and Revyva.
      </p>
      <p>
        I also enjoy playing disc golf, video games, and enjoying the local restraunts in the area.
      </p>
      <p>
        You may have noticed that almost all my work is in screenshot form - almost all of my professional work is on private repositories, so if you would like to know more, shoot me a message on my 
        <a
          href="https://www.linkedin.com/in/rachiu/"
          rel="noopener noreferrer"
          target="_blank"
        >
          LinkedIn
        </a>
      </p>
    </div>
  </section>
</DocumentFragment>
`;
